packages/thirdweb/src/wallets/in-app/core/lib/in-app-wallet-calls.ts

```diff
@@ -0,0 +1,99 @@
+import type { ThirdwebClient } from "../../../../client/client.js";
+import { eth_getTransactionReceipt } from "../../../../rpc/actions/eth_getTransactionReceipt.js";
+import { getRpcClient } from "../../../../rpc/rpc.js";
+import { sendAndConfirmTransaction } from "../../../../transaction/actions/send-and-confirm-transaction.js";
+import { sendBatchTransaction } from "../../../../transaction/actions/send-batch-transaction.js";
+import { type Hex, generateRandomHex } from "../../../../utils/encoding/hex.js";
+import type { PreparedSendCall } from "../../../eip5792/send-calls.js";
+import type {
+  GetCallsStatusResponse,
+  WalletCallReceipt,
+  WalletSendCallsId,
+} from "../../../eip5792/types.js";
+import type { Account, Wallet } from "../../../interfaces/wallet.js";
+
+const bundlesToTransactions = new Map<string, Hex[]>();
+
+/**
+ * @internal
+ */
+export async function inAppWalletSendCalls(args: {
```

Comment:
having the wallet.sendCalls in the interface would def make it much cleaner here. separate implementation for inapp vs smart. this function does 2 jobs implicitely here (smart + inapp)

Reply:
Would it be better to separate it anyway? They were originally separate but I combined them for reuse.